### PR TITLE
Feat: add styled tooltip and loading skeletons for the reputation chart

### PR DESCRIPTION
**Pull request by mmioana**
## Description

- [ ] We need to make use of the `ChartCustomTooltip` also for the reputation chart

- [ ] Add the loading skeletons for the component

[Figma component](https://www.figma.com/design/gac6xVgGNCAge5Bia933dp/Dashboard?node-id=4878-89693&node-type=frame&t=HxDBCcAqAILoneMc-0)

## Testing

TODO: Check ChartCustomTooltip is used for the pie chart present in the `Influence by team` or `Top contributors` card

* Step 1. Go to `planex`
* Step 2. Check `Influence by team` card is shown if no domain filtering is present
* Step 3. Hover one slice of the pie chart and check the tooltip

Screenshot 2024-09-18 at 11 06 01

* Step 4. Select one domain from the breadcrumbs dropdown (currently also `General` is going to show the `Top contributors` but this will be fixed in #3089)
* Step 5. Hover one slice of the pie chart and check the tooltip

Screenshot 2024-09-18 at 11 05 53

* Step 6. With the domain filtering active, hard refresh the page and check the loading spinners are showing

Screenshot 2024-09-18 at 11 11 08

## Diffs

**New stuff** ✨

* `ChartLoadingLayer` and `LegendLoadingItem`components

Resolves [#3109](https://github.com/JoinColony/colonyCDapp/issues/3109)

### CODE REVIEW

**Review by rumzledz — APPROVED** [review]
Awesome work on this @mmioana ✨ I can confirm that the chart's skeleton and tooltips are working 🙌 

reputation chart

File: src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/LegendLoadingItem.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import LoadingSkeleton from '~common/LoadingSkeleton/LoadingSkeleton.tsx';
+
+const displayName =
+  'v5.frame.ColonyHome.ReputationChart.partials.LegendLoadingItem';
+
+const LegendLoadingItem = () => {
+  return (
+    <div className="flex flex-row items-center gap-1">
+      <LoadingSkeleton className="h-2.5 w-2.5 rounded-full" isLoading />
```
[the remark this answers is not in the document]
Comment (mmioana) [reply src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/LegendLoadingItem.tsx:11]: Why I wouldn't do this change is only due to the fact we need to duplicate the `overflow-hidden skeleton` classes here and if we were to make some changes to how the `LoadingSkeleton` behaves or looks, we'll have to take care again wherever we want to use something similar 😢

File: src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/ChartLoadingLayer.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        />
+        <div
+          className={clsx(
+            absoluteCenteredClass,
+            'h-[102px] w-[102px] rounded-full bg-base-white',
+          )}
+        />
+      </div>
+    </div>
+  );
```
[the remark this answers is not in the document]
Comment (mmioana) [reply src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/ChartLoadingLayer.tsx:25]: Why I wouldn't do this change is only due to the fact we need to duplicate the overflow-hidden skeleton classes here and if we were to make some changes to how the LoadingSkeleton behaves or looks, we'll have to take care again wherever we want to use something similar 😢

File: src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/ChartLoadingLayer.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+import React from 'react';
+
+import LoadingSkeleton from '~common/LoadingSkeleton/LoadingSkeleton.tsx';
+import { tw } from '~utils/css/index.ts';
+
+const absoluteCenteredClass = tw`absolute left-1/2 top-1/2 z-base -translate-x-1/2 -translate-y-1/2`;
+
+export const ChartLoadingLayer = () => {
+  return (
+    <div className={absoluteCenteredClass}>
```
[the remark this answers is not in the document]
Comment (mmioana) [reply src/components/v5/frame/ColonyHome/partials/ReputationChart/partials/ChartLoadingLayer.tsx:11]: Only if we do `{isLoading ? <ChartLoadingLayer /> : <ResponsivePie ... />}`, but I preferred to keep it as an overlay to still get the animation when the data loads - try refreshing the page when a team is selected

**Comment by arrenv** [comment]
Just noting that the skeleton and list of teams/users should only ever be a max of two rows, to avoid the height jumping seen here on team filtering and on load.

**Comment by mmioana** [comment]
> Just noting that the skeleton and list of teams/users should only ever be a max of two rows, to avoid the height jumping seen here on team filtering and on load.

I'll create a separate issue for this one @arrenv 🙏 
As we might need to reconsider some aspects in the chart to only ever show 2 rows.
Now a question is popping up, what happens if we have very long team names?

**Comment by mmioana** [comment]
> > Just noting that the skeleton and list of teams/users should only ever be a max of two rows, to avoid the height jumping seen here on team filtering and on load.
> 
> I'll create a separate issue for this one @arrenv 🙏 As we might need to reconsider some aspects in the chart to only ever show 2 rows. Now a question is popping up, what happens if we have very long team names?

Found it in notion: `Should show a max character length of 20 characters. An ellipsis should be shown if the name exceeds this.`
I'll try to push a fix for this actually on this branch

**Review by rdig — APPROVED** [review]
Functions as advertised now. Very nice work! :100: 

No domain filter:

Screenshot from 2024-09-23 09-41-21

Domain filter:

Screenshot from 2024-09-23 09-41-47

Loader:

Screenshot from 2024-09-23 09-42-05

File: src/components/v5/frame/ColonyHome/partials/ReputationChart/ReputationChart.tsx [hunk 4]
```diff
@@ -36,7 +37,11 @@ const ReputationChart = () => {
     colony: { domains },
   } = useColonyContext();
 
-  const { contributorsList } = useContributorsByDomain();
+  const { contributorsList, loading: isContributorsLoading } =
+    useContributorsByDomain();
+
+  // @TODO here we should also check if the colony data is loading
+  const isDataLoading = isContributorsLoading;
```
Comment (iamsamgibbs) [inline src/components/v5/frame/ColonyHome/partials/ReputationChart/ReputationChart.tsx:44]: Is this TODO still needed?

**Review by iamsamgibbs — APPROVED** [review]
Great job with this, it all looks great!

Screenshot 2024-09-23 at 08 39 24

Screenshot 2024-09-23 at 08 39 39

Loading state matches figma perfectly:

Screenshot 2024-09-23 at 08 41 09

Also doesn't show on tablet as is intended:

Screenshot 2024-09-23 at 08 47 30

My only suggestion (and it's related to the bar chart but is only apparent now that this component is working) is to make the chart top aligned as it moves around when the number of team rows changes:
 
Screenshot 2024-09-23 at 08 43 56

Easier to see in video form:

https://github.com/user-attachments/assets/95eabfb0-7096-4066-9b94-df2322d50313

Just a suggestion though, take it or leave it! :)

File: src/components/v5/frame/ColonyHome/partials/ReputationChart/ReputationChart.tsx (hunk 4 above)
Comment (mmioana) [reply src/components/v5/frame/ColonyHome/partials/ReputationChart/ReputationChart.tsx:44]: I left this `TODO` only due to the fact the domain data comes from the colony context and by the way the code is implemented right now, we show the global loader until the colony data is fetched, hence it didn't really make sense to add the loading condition also here, at least for the moment

**Comment by mmioana** [comment]
> Great job with this, it all looks great!
> 
> Screenshot 2024-09-23 at 08 39 24 Screenshot 2024-09-23 at 08 39 39
> Loading state matches figma perfectly:
> 
> Screenshot 2024-09-23 at 08 41 09
> Also doesn't show on tablet as is intended:
> 
> Screenshot 2024-09-23 at 08 47 30
> My only suggestion (and it's related to the bar chart but is only apparent now that this component is working) is to make the chart top aligned as it moves around when the number of team rows changes:
> 
> Screenshot 2024-09-23 at 08 43 56
> Easier to see in video form:
> 
>  Screen.Recording.2024-09-23.at.08.43.38.mov 
> Just a suggestion though, take it or leave it! :)

I have removed the content full height from the `Total in and out` card, at least for the moment, so the chart and other sections no longer jump upon domain filtering 🙌  thank you for noticing this @iamsamgibbs 💯

**Pull request merged**